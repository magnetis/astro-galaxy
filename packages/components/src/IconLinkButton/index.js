import styled from 'styled-components';
import PropTypes from 'prop-types';
import { color, space, border, layout, fontSize, typography } from 'styled-system';
import { font, buttonSizes } from '@magnetis/astro-galaxy-core';
import { iconlabelMargin } from '../IconButton';

const IconLinkButton = styled.button`
  ${font};
  ${buttonSizes};

  display: inline-block;
  overflow: hidden;
  color: ${props => props.theme.colors.space100};
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  background-color: transparent;
  border: none;
  border-radius: 0;
  transition: all 0.3s ease;
  ${props => props.bold && 'font-weight: 700;'};

  & svg {
    margin: ${iconlabelMargin};
    vertical-align: middle;
    color: ${props => props.theme.colors.space100};
    transition: all 0.3s ease;
  }

  :hover:not(:disabled),
  :focus:not(:disabled) {
    color: ${props => props.theme.colors.uranus500};
    outline: none;

    & svg {
      color: ${props => props.theme.colors.uranus500};
    }
  }

  :active:not(:disabled) {
    color: ${props => props.theme.colors.uranus800};

    & svg {
      color: ${props => props.theme.colors.uranus800};
    }
  }

  :disabled {
    color: ${props => props.theme.colors.moon200};
    cursor: not-allowed;

    & svg {
      color: ${props => props.theme.colors.moon200};
    }
  }
`;

IconLinkButton.displayName = 'IconLinkButton';

IconLinkButton.defaultProps = {
  bold: false,
  iconlabel: true,
  iconlabelRight: false,
  font: 'secondary',
  buttonSize: 'medium',
};

IconLinkButton.propTypes = {
  bold: PropTypes.bool,
  iconlabel: PropTypes.bool,
  iconlabelRight: PropTypes.bool,
  buttonSize: PropTypes.string,
  font: PropTypes.string,
  ...color.propTypes,
  ...space.propTypes,
  ...border.propTypes,
  ...layout.propTypes,
  ...fontSize.propTypes,
  ...typography.propTypes,
};

export default IconLinkButton;
