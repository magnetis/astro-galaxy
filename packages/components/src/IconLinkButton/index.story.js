import React from 'react';
import { storiesOf } from '@storybook/react';
import { IconCalendar } from '@magnetis/astro-galaxy-icons';
import IconLinkButton from './index.js';

storiesOf('icon link button', module).add('default', () => (
  <div style={{ backgroundColor: '#32383c', height: '50vh' }}>
    <IconLinkButton>
      <IconCalendar />
      icon link
    </IconLinkButton>

    <IconLinkButton bold>
      <IconCalendar />
      icon link bold
    </IconLinkButton>

    <IconLinkButton iconlabel iconlabelRight>
      icon link
      <IconCalendar />
    </IconLinkButton>

    <IconLinkButton iconlabel iconlabelRight disabled>
      icon link
      <IconCalendar />
    </IconLinkButton>
  </div>
));
