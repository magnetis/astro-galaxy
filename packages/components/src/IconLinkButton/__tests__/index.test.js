import React from 'react';
import { IconCalendar } from '@magnetis/astro-galaxy-icons';
import IconLinkButton from '../index';

describe('IconLinkButton', () => {
  it('should render an icon link button with calendar icon', () => {
    const iconButton = rendererCreateWithTheme(
      <IconLinkButton>
        <IconCalendar />
        hello
      </IconLinkButton>
    ).toJSON();

    expect(iconButton).toMatchSnapshot();
  });

  it('should change font-weight when icon link button has bold prop', () => {
    const iconButton = rendererCreateWithTheme(
      <IconLinkButton bold>
        <IconCalendar />
        world
      </IconLinkButton>
    ).toJSON();

    expect(iconButton).toMatchSnapshot();
    expect(iconButton).toHaveStyleRule('font-weight', '700');
  });

  it('should render an icon link button with calendar icon on the right side', () => {
    const iconButton = rendererCreateWithTheme(
      <IconLinkButton iconlabel iconlabelRight>
        hello
        <IconCalendar />
      </IconLinkButton>
    ).toJSON();

    expect(iconButton).toMatchSnapshot();
  });
});
