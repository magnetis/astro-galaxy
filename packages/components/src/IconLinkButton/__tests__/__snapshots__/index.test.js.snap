// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLinkButton should change font-weight when icon link button has bold prop 1`] = `
.c1 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  padding: 8px 38px;
  font-size: 16px;
  line-height: 28px;
  border-radius: 24px;
  display: inline-block;
  overflow: hidden;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  background-color: transparent;
  border: none;
  border-radius: 0;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  font-weight: 700;
}

.c0 svg {
  margin: -4px 9px -3px 0;
  vertical-align: middle;
  color: #fff;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  color: #159ce4;
  outline: none;
}

.c0:hover:not(:disabled) svg,
.c0:focus:not(:disabled) svg {
  color: #159ce4;
}

.c0:active:not(:disabled) {
  color: #16518b;
}

.c0:active:not(:disabled) svg {
  color: #16518b;
}

.c0:disabled {
  color: #e9eef2;
  cursor: not-allowed;
}

.c0:disabled svg {
  color: #e9eef2;
}

<button
  className="c0"
>
  <svg
    className="c1"
    color="moon900"
    height={24}
    viewBox="0 0 32 32"
    width={24}
  >
    <defs>
      <path
        d="M16 4H8v2a1 1 0 1 1-2 0V4H3a1 1 0 0 0-1 1v4.111h20V5a1 1 0 0 0-1-1h-3v2a1 1 0 0 1-2 0V4zm1 0v2-2zm1-2v2h1V2h2a3 3 0 0 1 3 3v18a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h2v2h1V1a1 1 0 1 1 2 0v3h1V2h6v2h1V1a1 1 0 0 1 2 0v1zm-1 0V1v1zM7 4v2-2zm0-2V1v1zm15 9.111H2V23a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V11.111zM7 17a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        id="icon-calendar-a"
      />
    </defs>
    <g
      id="icon/secondary/calendar"
    >
      <g
        id="color/earth-400"
        transform="translate(4 3)"
      >
        <mask
          id="icon-calendar-b"
        >
          <use
            xlinkHref="#icon-calendar-a"
          />
        </mask>
        <use
          id="Mask"
          xlinkHref="#icon-calendar-a"
        />
        <g
          id="color/moon-900"
          mask="url(#icon-calendar-b)"
        >
          <path
            d="M0 0h32v32H0z"
            id="Rectangle"
            transform="translate(-4 -3)"
          />
        </g>
      </g>
    </g>
  </svg>
  world
</button>
`;

exports[`IconLinkButton should render an icon link button with calendar icon 1`] = `
.c1 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  padding: 8px 38px;
  font-size: 16px;
  line-height: 28px;
  border-radius: 24px;
  display: inline-block;
  overflow: hidden;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  background-color: transparent;
  border: none;
  border-radius: 0;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c0 svg {
  margin: -4px 9px -3px 0;
  vertical-align: middle;
  color: #fff;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  color: #159ce4;
  outline: none;
}

.c0:hover:not(:disabled) svg,
.c0:focus:not(:disabled) svg {
  color: #159ce4;
}

.c0:active:not(:disabled) {
  color: #16518b;
}

.c0:active:not(:disabled) svg {
  color: #16518b;
}

.c0:disabled {
  color: #e9eef2;
  cursor: not-allowed;
}

.c0:disabled svg {
  color: #e9eef2;
}

<button
  className="c0"
>
  <svg
    className="c1"
    color="moon900"
    height={24}
    viewBox="0 0 32 32"
    width={24}
  >
    <defs>
      <path
        d="M16 4H8v2a1 1 0 1 1-2 0V4H3a1 1 0 0 0-1 1v4.111h20V5a1 1 0 0 0-1-1h-3v2a1 1 0 0 1-2 0V4zm1 0v2-2zm1-2v2h1V2h2a3 3 0 0 1 3 3v18a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h2v2h1V1a1 1 0 1 1 2 0v3h1V2h6v2h1V1a1 1 0 0 1 2 0v1zm-1 0V1v1zM7 4v2-2zm0-2V1v1zm15 9.111H2V23a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V11.111zM7 17a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        id="icon-calendar-a"
      />
    </defs>
    <g
      id="icon/secondary/calendar"
    >
      <g
        id="color/earth-400"
        transform="translate(4 3)"
      >
        <mask
          id="icon-calendar-b"
        >
          <use
            xlinkHref="#icon-calendar-a"
          />
        </mask>
        <use
          id="Mask"
          xlinkHref="#icon-calendar-a"
        />
        <g
          id="color/moon-900"
          mask="url(#icon-calendar-b)"
        >
          <path
            d="M0 0h32v32H0z"
            id="Rectangle"
            transform="translate(-4 -3)"
          />
        </g>
      </g>
    </g>
  </svg>
  hello
</button>
`;

exports[`IconLinkButton should render an icon link button with calendar icon on the right side 1`] = `
.c1 {
  fill: currentcolor;
  color: #32383c;
}

.c0 {
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  padding: 8px 38px;
  font-size: 16px;
  line-height: 28px;
  border-radius: 24px;
  display: inline-block;
  overflow: hidden;
  color: #fff;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  background-color: transparent;
  border: none;
  border-radius: 0;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c0 svg {
  margin: -4px 0 -3px 9px;
  vertical-align: middle;
  color: #fff;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c0:hover:not(:disabled),
.c0:focus:not(:disabled) {
  color: #159ce4;
  outline: none;
}

.c0:hover:not(:disabled) svg,
.c0:focus:not(:disabled) svg {
  color: #159ce4;
}

.c0:active:not(:disabled) {
  color: #16518b;
}

.c0:active:not(:disabled) svg {
  color: #16518b;
}

.c0:disabled {
  color: #e9eef2;
  cursor: not-allowed;
}

.c0:disabled svg {
  color: #e9eef2;
}

<button
  className="c0"
>
  hello
  <svg
    className="c1"
    color="moon900"
    height={24}
    viewBox="0 0 32 32"
    width={24}
  >
    <defs>
      <path
        d="M16 4H8v2a1 1 0 1 1-2 0V4H3a1 1 0 0 0-1 1v4.111h20V5a1 1 0 0 0-1-1h-3v2a1 1 0 0 1-2 0V4zm1 0v2-2zm1-2v2h1V2h2a3 3 0 0 1 3 3v18a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V5a3 3 0 0 1 3-3h2v2h1V1a1 1 0 1 1 2 0v3h1V2h6v2h1V1a1 1 0 0 1 2 0v1zm-1 0V1v1zM7 4v2-2zm0-2V1v1zm15 9.111H2V23a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V11.111zM7 17a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm5-4a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
        id="icon-calendar-a"
      />
    </defs>
    <g
      id="icon/secondary/calendar"
    >
      <g
        id="color/earth-400"
        transform="translate(4 3)"
      >
        <mask
          id="icon-calendar-b"
        >
          <use
            xlinkHref="#icon-calendar-a"
          />
        </mask>
        <use
          id="Mask"
          xlinkHref="#icon-calendar-a"
        />
        <g
          id="color/moon-900"
          mask="url(#icon-calendar-b)"
        >
          <path
            d="M0 0h32v32H0z"
            id="Rectangle"
            transform="translate(-4 -3)"
          />
        </g>
      </g>
    </g>
  </svg>
</button>
`;
